finality-aleph/src/testing/data_store.rs

```diff
@@ -442,6 +448,38 @@ async fn sends_block_request_on_missing_block() {
     .await;
 }
 
+#[tokio::test]
+async fn sends_justification_request_when_not_finalized() {
+    run_test(|mut test_handler| async move {
+        let blocks = test_handler
+            .initialize_single_branch(MAX_DATA_BRANCH_LEN * 10)
+            .await;
+        test_handler.import_branch(blocks.clone()).await;
+
+        let blocks_branch = blocks[2..3].to_vec();
```

Comment:
my feeling is that `vec![blocks[2]]` is less confusing, feel free to ignore

Reply:
Fine, although it must be `vec![blocks[2].clone()]` for it to work.